### PR TITLE
autotune: simplify actuation

### PR DESCRIPTION
**Pull request by mlyle**
* always use the same "force", 6% of actuator scale.
The auto scale control has a relatively small useful range, and
sometimes does degenerate things.
* use a saner actuation counter mechanism, that is less likely to
do weird phase-modulatey things and is synchronous to the important
(sampling) clock.
* communicate phase of actuation counter through actuatordesired to
allow future work.

### CODE REVIEW

**Comment by ufoDziner** [comment]
Jenkins, build artifacts please.

**Comment by mlyle** [comment]
@ufoDziner reports this is flyable and does not catch fire.  But more work on this branch comin'.

**Comment by mlyle** [comment]
This is now up to the point that autotune-ng flight side could start getting hooked in.  However, there's been a lot of very dangerous refactor.  Requires careful review and bench test before flight test.

File: flight/Modules/Autotune/autotune.c [hunk 1]
```diff
@@ -209,13 +215,38 @@ static void at_new_actuators(UAVObjEvent * ev, void *ctx, void *obj, int len) {
...
+			running = true;
+			first_cycle = ! first_cycle;
+		}
+	}
+
+#ifdef AUTOTUNE_AVERAGING_MODE
+	struct at_measurement *avg_point = &at_averages[actuators->SystemIdentCycle];
+
+	if (first_cycle) {
+		bzero(avg_point, sizeof(*avg_point));
```
Comment (tracernz) [inline flight/Modules/Autotune/autotune.c]: bzero probably not a good idea for new code. Deprecated, and removed in newer POSIX standards. (https://linux.die.net/man/3/bzero)

**Review by tracernz — DISMISSED** [review]

**Comment by mlyle** [comment]
This is a good-sized chunk of code which is decent on its own for ongoing testing.  Setting ready for review.

**Comment by mlyle** [comment]
Test procedure for real vehicles:

An autotune should be run from default PIDs.  Then, email to mlyle@lyle.org:

1) a dump of the 'autotune' partition from the bootloader saved as vehiclename.bin   (go to firmware pane, click 'rescue', plug in board... then right click on autotune in partition browser, pick save, in 30 seconds provide a filename).  It's fine to zip or gzip this.
2) a uavo xml export after tune from the quad that contains the systemident object. (File menu - export UAV settings with the board connected *after tune*)
3) The githash (e.g. 8940fb9 ) of the build that you ran tests on.
4) Feedback on how the firmware-generated (old-style) autotune flies.

A variety of vehicles would be helpful.  Particularly interesting are vehicles that do not tune correctly currently (very low tau, or things like the Castle ESCs, etc), but we also need vehicles that came up with good autotunes for a baseline.

Also needed-- information on whether various targets work in bench test.

* [x] Normal F4 target (Sparky2)
* [x] OG Brain
* [x] Sparky
* [x] 1KHz F3 (DTFc, Lux)
* [x] BrainRE1

@mluessi if you are available to look at this I would be glad to enable for RE1 also-- I do not have cycles to test it right this moment myself as yet.

**Comment by mlyle** [comment]
jenkins, build artifacts please

**Comment by mlyle** [comment]
@ufoDziner has volunteered to test basic function on bench on RE1, so I have enabled it.

jenkins, build artifacts please    ( i need to delete the 'first' artifact build when it comes through )

**Comment by glowtape** [comment]
If two meters above grass qualifies as a bench, then the RE1 passed. Twice.

**Comment by ufoDziner** [comment]
RE1 bench test semi-successfully completed, and results emailed.  I couldn't apply the tune due to a low tau error.

**Comment by ufoDziner** [comment]
Just accomplished an autotune, and subsequent flight.

Autotune flight went well.  Seemed to be almost no delay after flipping the mode switch to when the oscillations started.  The oscillations seemed to be smaller, and more consistent.  Aside from that, it was as expected.

LoS flight went very well afterwards. The tune feels great, as did the tune from the old system.  Links to tunes are below.

New version: https://dronin-autotown.appspot.com/at/tune/ahFzfmRyb25pbi1hdXRvdG93bnIYCxILVHVuZVJlc3VsdHMYgICA4MWb_wkM

Old version: https://dronin-autotown.appspot.com/at/tune/ahFzfmRyb25pbi1hdXRvdG93bnIYCxILVHVuZVJlc3VsdHMYgICA4JLI0QkM

**Comment by pug398** [comment]
Testing on a Sparky. Tune went well with default settings and indoor flight seems ok. After initial tune https://dronin-autotown.appspot.com/at/tune/ahFzfmRyb25pbi1hdXRvdG93bnIYCxILVHVuZVJlc3VsdHMYgICA4OHvqQkM bumped damping/noise sliders to 0.9/1.3 for crisper feel. 

Robocat weight is 525g not 325g.

**Comment by mlyle** [comment]
@pug398 Awesome!  Can you please mail me the tune partition contents and the uavo file?  (The actual new stuff is mostly in that tune partition, which I still need to teach GCS to use--- though the other/current autotune should be made slightly better by this too)

**Comment by pug398** [comment]
Sure will dump them now and send.

File: flight/UAVObjects/uavobjectmanager.c [hunk 2]
```diff
@@ -1703,6 +1703,7 @@ static void invokeCallback(struct ObjectEventEntry *event, UAVObjEvent *msg,
 					// which might clobber r0-r3
 		: // no pure read-only registers
 		: "memory",		// callback may clobber memory,
+		"cc",			// condition codes,
```
Comment (tracernz) [inline flight/UAVObjects/uavobjectmanager.c:1706]: 👍

**Review by tracernz — APPROVED** [review]
LGTM 👍

**Comment by ufoDziner** [comment]
Flew a bunch of packs on the tunes from this, and have had quite good results.  No issues to report.

**Pull request merged**